src/components/VideoPlayer.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useContext } from 'react';
+import GlobalContext from '../state/GlobalContext';
+import Button from '@material-ui/core/Button';
+import styled from 'styled-components';
+
+const VideoPlayer = (props) => {
+    const {state, dispatch} = useContext(GlobalContext);
+    const {url, title, description, picture} = props;
+
+    const addToFavorites = () => {
+        dispatch({
+            type: 'ADD_FAVORITE_VIDEO',
+            payload: {
+                video: {
+                    url: url,
+                    title: title,
+                    description: description,
+                    picture: picture
+                }
+            }
+        });
+    }
+
+    const removeFromFavorites = () => {
+        dispatch({
+            type: 'REMOVE_FAVORITE_VIDEO',
+            payload: {
+                id: url,
+            }
+        });
+    }
+
+    const buttonForFavorites = () => {
+        if(state.user.loggedIn){
```

Comment:
try to refactor those nested ifs into single ifs.

https://refactoring.guru/replace-nested-conditional-with-guard-clauses